scripts/run_backend_tests.py

```diff
@@ -475,6 +496,9 @@ def main(args: Optional[List[str]] = None) -> None:
     """Run the tests."""
     parsed_args = _PARSER.parse_args(args=args)
 
+    install_third_party_libraries(
+        parsed_args.test_target, parsed_args.skip_install)
```

Comment:
It doesn't seem right for `install` method to accept the `skip_install` flag.

I would suggest having the `if skip_install` check here. Also, is there any specific reason that we can only have skip_install with test_target?

Reply:
> It doesn't seem right for install method to accept the skip_install flag.

It's okay. I tried to group all the installation dependency codes in one place. But we can also directly add the checks in `main()`

> I would suggest having the if skip_install check here. Also, is there any specific reason that we can only have skip_install with test_target?

It is mentioned in the issue description. Specifically, it is talking removal of installation of dependency removes performance overhead when running with `--test-target`

Reply:
We can add --skip_install and ask developers to use it if they want to skip the installation process. WDYT?

Reply:
> Do we want to also use this flag with any other flags? not just --test_target?

Yes, as there isn't a specific reason for using it only with `--test_target`. (And we have similar independent flags in other scripts as well.)

Reply:
@shricodev Is this thread resolved?